### PR TITLE
[MB-13572] Evaluation report PDF generation spike

### PR DESCRIPTION
**Pull request by mboren**
## [Jira ticket](https://dp3.atlassian.net/browse/MB-13572) for this change

Now when you click download report, you get a PDF.
It's been a while since anybody touched PDF generation, so the goal of this spike was to figure out how it works.

specific things I was testing here:
- we can generate a PDF using a template image
- we can put text into the PDF wherever we want
- we can ship send the generated PDF to the user's browser

I basically copied the code from `ShipmentSummaryWorksheet` with a few tweaks, and everything was fairly straightforward. No big surprises.

https://user-images.githubusercontent.com/2627844/188027283-ddd396b9-4bca-4ae9-baea-04f5f2f4a0d2.mov

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

### CODE REVIEW

File: pkg/paperwork/evaluation_report.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+var EvaluationReportPage1Layout = FormLayout{
+	TemplateImagePath: "pkg/paperwork/formtemplates/qae_spike_template.png",
+
+	// For now only lists a single shipment. Will need to update to accommodate multiple shipments
+	FieldsLayout: map[string]FieldPos{
+		"CUIBanner":        FormField(0, 1.5, 216, floatPtr(10), nil, stringPtr("CM")),
+		"ReportID":         FormField(80, 40, 216, floatPtr(10), nil, stringPtr("LM")),
+		"DateOfInspection": FormField(80, 60, 216, floatPtr(10), nil, stringPtr("LM")),
+		"EvaluationType":   FormField(80, 80, 216, floatPtr(10), nil, stringPtr("LM")),
+		"Remarks":          FormField(80, 105, 216, floatPtr(10), nil, stringPtr("LM")),
```
Comment (mboren) [inline pkg/paperwork/evaluation_report.go:21]: `FormField` lets us put text on top of the template image wherever we want.
```
func FormField(xPos, yPos, width float64, fontSize, lineHeight *float64, alignStr *string) FieldPos {
```

> Horizontal alignment is controlled by including "L", "C" or "R" (left, center, right) in alignStr.
> Vertical alignment is controlled by including "T", "M", "B" or "A" (top, middle, bottom, baseline) in alignStr.
> The default alignment is left middle.

File: scripts/gen-assets [hunk 2]
```diff
@@ -26,5 +26,5 @@ else
   echo "RUNNING LOCALLY"
   image=milmove/circleci-docker:milmove-app-9ea16004ab38c5e43e715e09241515ec220b6067
   docker pull -q "${image}"
-  docker run -it --rm=true -v "${PWD}:${PWD}" -w "${PWD}" "${image}" go-bindata -modtime "${modtime}" -o "${output_file}" -pkg "${package_name}" "${input_dirs[@]}"
+  docker run --platform linux/amd64 -it --rm=true -v "${PWD}:${PWD}" -w "${PWD}" "${image}" go-bindata -modtime "${modtime}" -o "${output_file}" -pkg "${package_name}" "${input_dirs[@]}"
```
Comment (mboren) [inline scripts/gen-assets:29]: i had to add `--platform linux/amd64` to get this to run. It's apparently an M1 thing. Would love it if somebody with an older (non-M1) mac could run this version of the script to see if there are any problems.

File: src/components/Office/EvaluationReportTable/EvaluationReportTable.jsx [hunk 3]
```diff
@@ -21,6 +21,23 @@ const EvaluationReportTable = ({ reports, emptyText, moveCode, customerInfo, gra
...
+      if (win) {
+        win.opener = null;
+        win.location = url;
+      }
+    }
+  }
+
+  const handleDownloadReportClick = (reportID) => {
+    return safeOpenInNewTab(`/ghc/v1/evaluation-reports/${reportID}/download`);
+  };
```
Comment (mboren) [inline src/components/Office/EvaluationReportTable/EvaluationReportTable.jsx:40]: i have no idea what we want to do with the PDF when you click download. Should it open in the browser? should it just start the download? so I just copied what we were using for the shipment summary worksheet. I'm assuming we'll want to change this.

File: src/components/Office/EvaluationReportTable/EvaluationReportTable.jsx [hunk 4]
```diff
@@ -46,7 +63,9 @@ const EvaluationReportTable = ({ reports, emptyText, moveCode, customerInfo, gra
           {!report.submittedAt && <a href={`${location.pathname}/${report.id}`}>Edit report</a>}
         </td>
         <td className={styles.downloadColumn}>
-          <a href={`${location}/evaluation-reports/${report.id}/download`}>Download</a>
+          <Button className="usa-button--unstyled" onClick={() => handleDownloadReportClick(report.id)}>
+            Download
+          </Button>
```
Comment (mboren) [inline src/components/Office/EvaluationReportTable/EvaluationReportTable.jsx:68]: changing this to a button changes formatting a little by by shifting the link to the right. i just made this change for convenience, there's probably a better way

Comment (mboren) [reply src/components/Office/EvaluationReportTable/EvaluationReportTable.jsx:68]: also this breaks some client tests that are querying for this link

**Review by mandaem — APPROVED** [review]

**Comment by PDickeyTruss** [comment]
✅ Works as expected, PDF is generated with content related to the eval report.  Very helpful comments in this PR highlighting key aspects of PDF generation. Seems reasonable to me this could be extended to create a PDF containing all eval report info. 

One thought comes to mind: The generated PDF will look quite a bit different than what is shown in the browser when viewing a report (layout/styles/fonts/etc). I don't necessarily think this is an issue (I am not familiar with the PDF designs). However, if styling and/or matching what is shown in browser is important, it may be worth revisiting this and trying out some front-end PDF generation packages in hopes of re-using existing frontend work for PDF generation.

File: scripts/gen-assets (hunk 2 above)
Comment (PDickeyTruss) [reply scripts/gen-assets:29]: I am on M1, so can't give ya good data here. Will confirm it worked with no warnings for me.

**Pull request closed without merging**